In [55]:
import pandas as pd
import seaborn as sns

In [56]:
sns.get_dataset_names()

['anagrams',
 'anscombe',
 'attention',
 'brain_networks',
 'car_crashes',
 'diamonds',
 'dots',
 'dowjones',
 'exercise',
 'flights',
 'fmri',
 'geyser',
 'glue',
 'healthexp',
 'iris',
 'mpg',
 'penguins',
 'planets',
 'seaice',
 'taxis',
 'tips',
 'titanic']

In [57]:
df = sns.load_dataset('penguins')
df.head()

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
0,Adelie,Torgersen,39.1,18.7,181.0,3750.0,Male
1,Adelie,Torgersen,39.5,17.4,186.0,3800.0,Female
2,Adelie,Torgersen,40.3,18.0,195.0,3250.0,Female
3,Adelie,Torgersen,NaN,NaN,NaN,NaN,NaN
4,Adelie,Torgersen,36.7,19.3,193.0,3450.0,Female


In [58]:
df.shape

(344, 7)

There is 344 rows and 7 columns in the penguins dataset

In [59]:
df.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex'],
      dtype='object')

# Columns Description

Species: Denotes penguin species(Adelie, Chinstrap, Gento)

Island: Isalndas penguine found

Bill_length: Bill length of penguines (mm)

Bill_depth : Bill depth of penguines (mm)

flipper_length: flipper length of penguines (mm)

body_mass_g: It the weight of the penguin measured in grams (g).

Sex: male, female


# Target column is species

# Data Analysis 

In [60]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
dtypes: float64(4), object(3)
memory usage: 18.9+ KB


In [61]:
df.duplicated().sum()

np.int64(0)

In [62]:
df.isna().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

In [63]:
df1 = df.copy()
df2 = df.copy()
df3 = df.copy()
df4 = df.copy()

In [64]:
df['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [65]:
df['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

There is three category in island.

Where Biscoe has the most penguine(168) and Torgersen has the lowest 52

In [66]:
df.groupby('species')['sex'].value_counts()

species    sex   
Adelie     Female    73
           Male      73
Chinstrap  Female    34
           Male      34
Gentoo     Male      61
           Female    58
Name: count, dtype: int64

Every species has almost equal amount of male and female

In [67]:
df.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex'],
      dtype='object')

In [68]:
df.groupby('species')['island'].value_counts()

species    island   
Adelie     Dream         56
           Torgersen     52
           Biscoe        44
Chinstrap  Dream         68
Gentoo     Biscoe       124
Name: count, dtype: int64

Adelie species is found in every island. But Chinstrap are found only in Dream island. Gentoo found in only Biscoe islan

In [69]:
df.duplicated().sum()

np.int64(0)

No duplicate row in the dataset

# Handling the nan values

In [70]:
df.isna().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

biil_length, bill_depth, flipper_length and body_mass , all have 2 nan values and sex column has 11 nan values.

In [72]:
df['bill_length_mm'] = df['bill_length_mm'].fillna(df['bill_length_mm'].mean())

In [76]:
mean_depth = df['bill_depth_mm'].mean()

df['bill_depth_mm'] = df['bill_depth_mm'].fillna(mean_depth)

In [78]:
mean_flipper = df['flipper_length_mm'].mean()

df['flipper_length_mm'] = df['flipper_length_mm'].fillna(mean_flipper)

In [79]:
mean_mass = df['body_mass_g'].mean()

df['body_mass_g'] = df['body_mass_g'].fillna(mean_mass)

In [81]:
df['sex'].value_counts()

sex
Male      168
Female    165
Name: count, dtype: int64

In [83]:
mode_sex = df['sex'].mode()[0]

df['sex'] = df['sex'].fillna(mode_sex)

In [84]:
df.isna().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
dtype: int64

In [86]:
df.shape

(344, 7)